importApp/src/main/java/com/commercetools/dataimport/products/ProductDraftReader.java

```diff
@@ -0,0 +1,313 @@
+                setStrict(false);
+            }});
+            setFieldSetMapper(new PassThroughFieldSetMapper());
+        }};
+        FlatFileItemReader<FieldSet> reader = new FlatFileItemReader<>();
+        reader.setResource(this.attributeDefinitionsCsvResource);
+        reader.setLineMapper(fullLineMapper);
+        reader.setLinesToSkip(1);
+        this.delegate = reader;
+    }
+
+    @Override
+    public ProductDraft read() throws Exception {
+        if (currentProducts < maxProducts) {
+            currentProducts++;
+            return readDelegate();
+        }
+        return null;
+    }
+
+    private ProductDraft readDelegate() throws Exception {
+        ProductDraftBuilder entry = null;
+        FieldSet currentLine = null;
+        do {
+            currentLine = delegate.read();
+            if (currentLine != null) {
+                if (isNewEntry(currentLine)) {
+                    if (prevEntry != null) {
+                        entry = prevEntry;
+                        prevEntry = createNewEntry(currentLine);
+                    } else {
+                        prevEntry = createNewEntry(currentLine);
+                    }
+                } else {
+                    final ProductVariantDraftBuilder variantDraftBuilder = createNewVariantDraftBuilder(currentLine, prevEntry.build().getProductType().getKey());
+                    prevEntry.plusVariants(variantDraftBuilder.build());
+                }
+            }
+        } while (currentLine != null && entry == null);
+        return entry != null ? entry.build() : null;
+    }
+
+    private ProductVariantDraftBuilder createNewVariantDraftBuilder(final FieldSet currentLine, final String productTypeKey) throws BindException {
+        final ProductsCsvEntry productsCsvEntry = mapLineToEntry(currentLine);
+        final ProductVariantDraftBuilder builder = ProductVariantDraftBuilder.of();
+        builder.sku(productsCsvEntry.getSku());
+        builder.prices(parsePricesLine(productsCsvEntry.getPrices()));
+        builder.attributes(parseAttributes(currentLine, productTypes, productTypeKey));
+        addImages(productsCsvEntry, builder);
+        return builder;
+    }
+
+    private void addImages(final ProductsCsvEntry productsCsvEntry, final ProductVariantDraftBuilder builder) {
+        if (productsCsvEntry.getImages().contains(";") || productsCsvEntry.getImages().contains(",")) {
+            throw new RuntimeException("not prepared for multi images" + productsCsvEntry.getImages());
+        }
+        if (!isEmpty(productsCsvEntry.getImages())) {
+            final String url = productsCsvEntry.getImages();
+            builder.images(Image.of(url, ImageDimensions.of(0, 0)));
+        }
+    }
+
+    private List<AttributeDraft> parseAttributes(final FieldSet currentLine, final List<ProductType> productTypes, final String productTypeKey) {
```

Comment:
That's some massive method :D
Could you split it? Maybe it's already planned for the next improvements, so no need to do it in this PR.